cmd/write_graph.go

```diff
@@ -0,0 +1,205 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"github.com/arangodb/go-driver"
+  "github.com/neunhoef/collectionmaker/pkg/database"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"math/rand"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+)
+
+var (
+	cmdWriteGraph = &cobra.Command{
+		Use:   "graph",
+		Short: "Write vertices and edges and update them",
+		RunE:  writeGraph,
+	}
+)
+
+// This uses the types Instance and Step from "create_graph.go"
+
+func init() {
+	var parallelism int = 1
+	var startDelay int64 = 5
+	var number int64 = 1000000
+	cmdWriteGraph.Flags().IntVar(&parallelism, "parallelism", parallelism, "set -parallelism to use multiple go routines")
+	cmdWriteGraph.Flags().Int64Var(&number, "number", number, "set -number for number of edges to write per go routine")
+	cmdWriteGraph.Flags().Int64Var(&startDelay, "start-delay", startDelay, "Delay between the start of two go routines.")
+}
+
+// writeGraph writes edges in parallel
+func writeGraph(cmd *cobra.Command, _ []string) error {
+	parallelism, _ := cmd.Flags().GetInt("parallelism")
+	number, _ := cmd.Flags().GetInt64("number")
+	startDelay, _ := cmd.Flags().GetInt64("start-delay")
+
+	db, err := _client.Database(context.Background(), "_system")
+	if err != nil {
+		return errors.Wrapf(err, "can not get database: %s", "_system")
+	}
+
+	if err := writeGraphParallel(parallelism, number, startDelay, db); err != nil {
+		return errors.Wrapf(err, "can not setup some tenants")
+	}
+
+	return nil
+}
+
+// writeSomeGraphParallel creates some edges in parallel
+func writeGraphParallel(parallelism int, number int64, startDelay int64, db driver.Database) error {
+	var mutex sync.Mutex
+	totaltimestart := time.Now()
+	wg := sync.WaitGroup{}
+	haveError := false
+	for i := 1; i <= parallelism; i++ {
+	  time.Sleep(time.Duration(startDelay) * time.Millisecond)
+		i := i // bring into scope
```

Comment:
that line is not necessary because `i` value is passed to the GOroutine function and inside the function, there is a copy of it.